pkg/types/alibabacloud/validation/platform.go

```diff
@@ -17,11 +17,46 @@ func ValidatePlatform(p *alibabacloud.Platform, n *types.Networking, fldPath *fi
 		allErrs = append(allErrs, field.Required(fldPath.Child("region"), "region must be specified"))
 	}
 
+	if p.PrivateZoneID != "" {
+		if p.VpcID == "" {
+			allErrs = append(allErrs, field.Required(fldPath.Child("vpcID"), "when using existing privatezones, an existing VPC must be used"))
+		}
+	}
+
+	if len(p.VSwitchIDs) > 0 {
+		allErrs = append(allErrs, validateVSwitches(p, fldPath)...)
+	}
+
 	allErrs = append(allErrs, validateMetadataServerIP(n)...)
 
 	return allErrs
 }
 
+func validateVSwitches(p *alibabacloud.Platform, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p.VpcID == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("vpcID"), "when using existing VSwitches, an existing VPC must be used"))
+	}
+
+	vswitchIDs := []string{}
+	for _, vswitchID := range p.VSwitchIDs {
+		isRepeated := false
+		for _, id := range vswitchIDs {
+			if vswitchID == id {
+				isRepeated = true
+				break
+			}
+		}
+		if isRepeated {
+			allErrs = append(allErrs, field.Duplicate(fldPath.Child("vswitchIDs"), vswitchID))
```

Comment:
Include the index to the duplicate in the field path.

Reply:
Still need to include the index in the field path.
```suggestion
			allErrs = append(allErrs, field.Duplicate(fldPath.Child("vswitchIDs").Index(i), vswitchID))
```